File: 0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int n = chars.size();
        int i = 0,res = 0;
        while(i < n) {
            int sameChar = 1;
            while(i + sameChar < n && chars[i] == chars[i+sameChar]) {
                sameChar++;
            }
            chars[res++] = chars[i];
            if(sameChar > 1) {
                for(char ch : to_string(sameChar)) {
                    chars[res++] = ch;
                }
            }
            i += sameChar;
        }
        return res;
    }
};

File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] chars) {
       int low = 0, res = 0;
        while(low < chars.length) {
            int sameChar = 1;
            while( low + sameChar < chars.length && chars[low] == chars[low + sameChar]) {
                sameChar++;
            }
            chars[res++] = chars[low];
            if(sameChar > 1) {
                for(char ch : Integer.toString(sameChar).toCharArray()) {
                chars[res++] = ch;
                }
            }
            
            low += sameChar;
        }
        return res;
    }
}